src/framework/mpas_io.F

```diff
@@ -6498,6 +6531,10 @@ subroutine MPAS_io_err_mesg(ioContext, ierr, fatal)
             call mpas_log_write('MPAS IO Error: Would clobber existing file', MPAS_LOG_ERR)
          case (MPAS_IO_ERR_NOEXIST_READ)
             call mpas_log_write('MPAS IO Error: Attempting to read a file which does not exist.', MPAS_LOG_ERR)
+         case (MPAS_IO_ERR_MISSING_DIM)
+            call mpas_log_write('MPAS IO Error: Attempting to read a dimension which does not exist.', MPAS_LOG_ERR)
+         case (MPAS_IO_ERR_INSUFFICIENT_ARG)
+            call mpas_log_write('MPAS IO Error: Attempting to read a string into a buffer which is too small.', MPAS_LOG_ERR)
```

Comment:
Just a thought, but what if we named this error code `MPAS_IO_ERR_INSUFFICIENT_BUF`, and generalized the error string to something like: `MPAS IO Error: Attempting to read a variable into a buffer of insufficient size.` This might allow us to re-use this error condition more generally throughout the `mpas_io` module when we detect that there is too much data for the buffer that was provided, rather than having the error be specific to strings.